src/krkn_lib/tests/test_krkn_kubernetes_monitor.py

```diff
@@ -184,20 +189,30 @@ def test_pods_by_label_never_respawn(self):
         self.deploy_namespace(namespace, [])
         self.deploy_delayed_readiness_pod(delayed_1, namespace, 0, label)
         self.deploy_delayed_readiness_pod(delayed_2, namespace, 0, label)
-        monitor_timeout = 10
-        time.sleep(5)
+        self.wait_pod(delayed_1, namespace)
+        self.wait_pod(delayed_2, namespace)
+
+        monitor_timeout = 15
+
         pods_and_namespaces = self.lib_k8s.select_pods_by_label(
-            f"test={label}"
+            f"test={label}", field_selector="status.phase=Running"
         )
         pods_thread = self.lib_k8s.monitor_pods_by_label(
-            f"test={label}", pods_and_namespaces, monitor_timeout
+            f"test={label}",
+            pods_and_namespaces,
+            field_selector="status.phase=Running",
+            max_timeout=monitor_timeout,
         )
-
         self.background_delete_pod(delayed_1, namespace)
+        time.sleep(3)
         result = pods_thread.join()
+        print("result " + str(result.unrecovered) + str(result.recovered))
+        if len(result.unrecovered) > 0:
+            print("unrecover" + str(result.unrecovered[0].__dict__))
+        if len(result.recovered) > 0:
+            print("recover" + str(result.recovered[0].__dict__))
         self.assertIsNotNone(result.error)
```

Comment:
@tsebastiani are you able to take a look at this test (I think it will fail). I can't seem to get it to error. But I'm not sure in what case it would show an error vs no unrecovered or recovered pods